pytest_operator/plugin.py

```diff
@@ -500,37 +732,83 @@ async def create_crash_dump(self) -> bool:
             log.info("juju-crashdump command was not found.")
             return False
 
-    async def _cleanup_model(self):
-        if not self.model:
+    async def _model_gone(self, model_name: Optional[str]):
+        if not model_name or not self._controller:
             return
+        models = await self._controller.model_uuids()
+        while model_name in models:
+            await asyncio.sleep(5.0)
+            models = await self._controller.model_uuids()
 
-        await self.log_model()
+    async def forget_model(
```

Comment:
Does this work if we do `ops_test.forget_model('main')`?  What model will we end up with as the current context?

Reply:
this happens during `_cleanup_models` actually when the module scope fixture `ops_test` is reaped by pytest.  

So if you forget any model's alias, and that model was the `current_alias`, then `ops_test.current_alias` is set to `None`

Reply:
ok and that then addresses the use case for `alias` being able to be set to `None` (asked [here]](https://github.com/charmed-kubernetes/pytest-operator/pull/69/files/774512c6de06d015c81353d639ef2af6b17f73fe#r846436036)).  Still feels a bit odd to me that it can be set to `None` by a user, but at least blocking it to ever be `None` seems wrong based on the code here